LiteDB/Engine/Disk/DiskService.cs

```diff
@@ -171,40 +162,55 @@ public PageBuffer NewPage()
         }
 
         /// <summary>
-        /// Write pages inside file origin using async queue - WORKS ONLY FOR LOG FILE - returns how many pages are inside "pages"
+        /// Write all pages inside log file in a thread safe operation
         /// </summary>
-        public int WriteAsync(IEnumerable<PageBuffer> pages)
+        public int WriteLogDisk(IEnumerable<PageBuffer> pages)
         {
             var count = 0;
+            var stream = _writer.Value;
 
-            foreach (var page in pages)
+            // do a global write lock - only 1 thread can write on disk at time
+            lock(stream)
```

Comment:
A minor thing - should be ok here, but usually it is better to have a separate object for locking